openfisca_core/simulations.py

```diff
@@ -297,3 +474,11 @@ def __init__(self, path, message, code = None):
         dpath.util.new(self.error, dpath_path, message)
         self.code = code
         Exception.__init__(self, self.error)
+
+
+def calculate_output_add(simulation, variable_name, period):
```

Comment:
> This is a dirty hook, as far a I know only used for decompositions.
>  `simulation.calculate_output` can be less strict about periods than `simulation.calculate` for a variable is a `calculate_output` has been set as a variable attribute.

Reply:
A [hook](https://stackoverflow.com/questions/467557/what-is-meant-by-the-term-hook-in-programming) is a way of bringing custom behiavior in the execution of the program.

I'd call this one dirty because:
- It is not documented
- The naming is not expressive
- Instead of multiplying these kinds of tricks, I'd prefer trying to understand more thoroughly the user needs behind it, and find a solution that is more meaningful.

Reply:
Should we open an issue about it ?

Reply:
Sure, something along "Document or deprecate calculate_output hooks"